core/shared/src/main/scala/result.scala

```diff
@@ -56,34 +61,126 @@ sealed class Result[+T, E <: Exception](val answer: T, val errors: Seq[(ClassTag
       val probs = res.errors ++ errors
       Result[T2, E with E2](res.get, probs)
     } catch {
-      case e: NullPointerException if !errors.isEmpty => Errata[T2, E with E2](errors)
+      case e: NullPointerException if errors.nonEmpty => Errata[T2, E with E2](errors)
       case e: Throwable => Unforeseen[T2, E with E2](e)
     }
 
   def map[T2](fn: T => T2) = Result[T2, E](fn(answer), errors)
 
-  def resolve[E2, T2 >: T](handlers: Each[E2, T2]*)(implicit ev: E2 <:< E): Resolved[T2, Nothing] = this match {
```

Comment:
We need this constraint here, otherwise it's possible to `resolve` a `Result` without handling all cases.